lib/semian/resource.rb

```diff
@@ -4,8 +4,8 @@ class Resource #:nodoc:
 
     class << Semian::Resource
       # Ensure that there can only be one resource of a given type
-      def instance(*args)
-        Semian.resources[args.first] ||= ProtectedResource.new(args.first, new(*args), nil)
+      def instance(*args, **kwargs)
+        Semian.resources[args.first] ||= ProtectedResource.new(args.first, new(args.first, **kwargs), nil)
```

Comment:
There's still something that isn't quite right here.

You grab all positional arguments with `*args` but only pass the first one.

So `instance(1, 2, 3, foo: 42)` works, but `2, 3` are simply ignored.

If you know that instance only takes a single positional argument, then the proper signature is `(name, **kwargs)`

Reply:
That's fair, this was just what the old code had, I'll fix it now